pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -802,7 +806,7 @@ func (r *ReconcileClusterPool) reconcileDeletedPool(pool *hivev1.ClusterPool, lo
 	if err != nil {
 		return err
 	}
-	for _, cd := range append(cds.Assignable(), cds.Installing()...) {
+	for _, cd := range cds.Unassigned(true) {
```

Comment:
Why weren't considering broken cds before that we are now?

Reply:
I believe that was likely a latent bug -- this code path was missed when we introduced the concept of broken clusters.

We invoke `reconcileDeletedPool` early in the reconcile, before we get a whack at cleaning up the broken clusters, and the code path that does that never gets any further if the pool is marked for deletion. So I think if you timed the deletion just right, you could have left zombie broken clusters around "forever".